### PR TITLE
EVG-3525: restrict parent hosts from spawning tasks and starting agents

### PR DESCRIPTION
**Pull request by li-cindy**

### CODE REVIEW

File: model/host/db.go [hunk 1]
```diff
@@ -398,6 +398,7 @@ func NeedsNewAgent(currentTime time.Time) db.Q {
 	return db.Query(bson.M{
 		StatusKey:    evergreen.HostRunning,
 		StartedByKey: evergreen.User,
+		bsonutil.GetDottedKeyName(DistroKey, distro.NoTasksAllowedKey): false,
```
Comment (john-m-liu) [inline model/host/db.go]: This should be NoTasksAllowed: ($ne: true} because you marked the field as omitempty

File: model/distro/distro.go [hunk 2]
```diff
@@ -30,9 +30,10 @@ type Distro struct {
 	SSHKey      string   `bson:"ssh_key,omitempty" json:"ssh_key,omitempty" mapstructure:"ssh_key,omitempty"`
 	SSHOptions  []string `bson:"ssh_options,omitempty" json:"ssh_options,omitempty" mapstructure:"ssh_options,omitempty"`
 
-	SpawnAllowed bool        `bson:"spawn_allowed" json:"spawn_allowed,omitempty" mapstructure:"spawn_allowed,omitempty"`
-	Expansions   []Expansion `bson:"expansions,omitempty" json:"expansions,omitempty" mapstructure:"expansions,omitempty"`
-	Disabled     bool        `bson:"disabled,omitempty" json:"disabled,omitempty" mapstructure:"disabled,omitempty"`
+	SpawnAllowed   bool        `bson:"spawn_allowed" json:"spawn_allowed,omitempty" mapstructure:"spawn_allowed,omitempty"`
+	NoTasksAllowed bool        `bson:"no_tasks_allowed" json:"no_tasks_allowed,omitempty" mapstructure:"no_tasks_allowed,omitempty"`
```
Comment (john-m-liu) [inline model/distro/distro.go]: I don't think we need a new field - you can do the same thing by checking the HasContainers field in the host document

Comment (li-cindy) [reply model/distro/distro.go]: Do I have access to the host document fields when I'm working in `distros.html`? I'm trying to think of how to set `HasContainers` via a checkbox.

Comment (john-m-liu) [reply model/distro/distro.go]: No, but my point is that this field isn't needed so you don't need to set it. The only change needed is the NeedsNewAgent query I think

Comment (li-cindy) [reply model/distro/distro.go]: Removed new field and am now using `HasContainers` in the query

File: service/templates/distros.html [hunk 3]
```diff
@@ -52,6 +53,7 @@ <h2 style="display:inline-block; padding-right:15px">Configure
       <input required ng-readonly="!activeDistro.new" id="identifier" name="id" type="text" class="form-control" ng-model="activeDistro._id" placeholder="Unique identifier for this distro">
       <label class="icon fa fa-warning distro-error" ng-show="form.id.$error.required">Distro identifier is required<br></label>
       <label class="icon fa fa-warning distro-error" ng-show="form.id.$dirty && form.id.$error.unsique || (activeDistro.new && form.id.$error.unique)">Distro identifier already exists</label>
+      <label class="icon fa fa-warning" ng-show="containerPoolDistros.indexOf(activeDistro._id) >= 0">Distro is not useable for tasks.</label>
```
Comment (bsamek) [inline service/templates/distros.html]: Could you add a why? Non-admins can see the distro page, and they won't have context to understand this message.

**Pull request merged**